### DISCUSSION TITLE
Bitdefender Necessary Exceptions

### DISCUSSION THREAD

**Discussion by ninjamonkey198206** [amidaware/tacticalrmm#769] [Show and tell]
I was having issues using the exceptions listed in the documentation when using Bitdefender Total Security (or any Bitdefender product with their Advanced Threat Defense). I'm still testing things out so I'm using the unsigned version of the agent.

The issue is that they limit exclusions by type, and don't allow * to be used for certain types, so you'll need to specify every single .exe file for Advanced Threat Defense. Make sure to select ALL available choices when entering the exceptions, A/V, ATD, and/or OTP.

In addition to the exceptions listed in the documentation, the following are necessary for Bitdefender, though more may be needed depending on use, so please feel free to add to the list:
```text
C:\Program Files\Mesh Agent\MeshAgent.exe
C:\Program Files\TacticalAgent\meshagent.exe
C:\Program Files\TacticalAgent\nssm.exe
C:\Program Files\TacticalAgent\tacticalrmm.exe
C:\Program Files\TacticalAgent\unins000.exe
```
Make sure to add exclusions for the api, rmm, and mesh urls as well, the actual installer app before you place it on the system, and to go back after the installation and edit the exceptions to include all available types and options (A/V, OTP, ATD).

Oh, they're also obnoxious and don't allow right click mouse interaction with the entry field, so you'll need to use ctrl-c and ctrl-v to save time typing those out.

This should ensure that Bitdefender won't cause any trouble for you.

**Comment by ninjamonkey198206** [comment]
## Bitdefender Total Security exceptions

Open the app and select the Protection tab on the left.

![Screenshot 2022-04-08 101539](https://user-images.githubusercontent.com/24654529/162472102-4419b177-dbe7-412e-a1a6-8189d1b5b1c0.png)

Open Antivirus, Advanced Threat Defense, or Online Threat Protection. It doesn't matter which, simply select the Settings tab in any of them and click Manage exceptions.

Click the Add an Exception button.

![Screenshot 2022-04-08 102200](https://user-images.githubusercontent.com/24654529/162472884-cdcdddc7-8000-47f8-8708-aacf1038d0df.png)

You'll be greeted with this panel.

![Screenshot 2022-04-08 102337](https://user-images.githubusercontent.com/24654529/162473180-48ab15a0-f74f-42bb-af0a-6225da84dd9e.png)

The buttons become available depending on what you enter for the path/URL. Antivirus allows folders and files. Online Threat Prevention allows URLS, files, folders, and *. Advanced Threat Defense allows only individual files.

Bitdefender does not allow right click interactions in the entry field, so copy and paste using ctrl-c and ctrl-v in you want to do so.

Enter
```text
C:\Program Files\Mesh Agent\MeshAgent.exe
```
and toggle the buttons for Advanced Threat Defense, Online Threat Prevention, and Antivirus. Under Antivirus, options for On-access scan, On-demand scan, and Embedded scripts appear. Select them all.

![Screenshot 2022-04-08 103322](https://user-images.githubusercontent.com/24654529/162474944-bda384a5-5569-4760-a4da-ffdac48b7ebc.png)

Repeat this process for
```text
C:\Program Files\TacticalAgent\meshagent.exe
C:\Program Files\TacticalAgent\tacticalrmm.exe
C:\Program Files\TacticalAgent\unins000.exe
```
Next, add an exception for
```text
C:\Windows\temp\trmm
```
Select all available options

![Screenshot 2022-04-08 103856](https://user-images.githubusercontent.com/24654529/162475827-271e307a-d916-454e-910d-34b05b316d41.png)

For the sake of caution, add the following and select the Online Threat Prevention option
```text
C:\Program Files\Mesh Agent\*
C:\Program Files\TacticalAgent\*
C:\Program Files\TacticalAgent\nssm.exe
C:\Windows\Temp\winagent-v*.exe
C:\temp\tacticalrmm*.exe
```
Add exceptions for your T-RMM URLs as well.

T-RMM seems to create other folders in C:\Windows\Temp as well, but so far no problems have come up from this. Do not exclude the entire Temp folder.

That should be all that's needed, unless Bitdefender complains about the T-RMM agent installation file, in which case you'll need to add an exception for it as well, selecting all available options, then redownload it and save it in the same location to be safe.

**Comment by silversword411** [comment]
Thank you, that was a great writeup that just took some copy/pasting :)

https://github.com/amidaware/trmm-docs/commit/9f1883ce36db4a24eb4d2d79da3421a379b71051

**Comment by ninjamonkey198206** [comment]
No problem.

Just noticed this, though:

Bitdefender does not allow right click interactions in the entry field, so copy and paste using ctrl-c and ctrl-v in you want to do so.

"ctrl-v in" should be "ctrl-v if"

Always has to be one little mistake...
